### ISSUE TITLE
test_expanded_reduction_cpu failure on pytorch/pytorch CI

### ISSUE THREAD

**Issue by jansel** [pytorch/torchdynamo#1697]

**Comment by jansel** [comment]
Looks like `test_dense_mask_index` might have a similar issue.  

I disabled the two tests in https://github.com/pytorch/pytorch/pull/87084/files

To reproduce you can enable the two tests and add the `ciflow/periodic` to your PR.

**Comment by zhuhaozhe** [comment]
Hi, @jansel and @jgong5. I have investigated these two UTs:
```
test_dense_mask_index_cpu (__main__.CpuTests)
test_expanded_reduction_cpu (__main__.CpuTests)
```
They should be caused by "difference reduce order" for "sum". The "add" operator between floating points needs to align their exponentiation and will lose accuracy here. Different order to "sum" means different accuracy loss.

For "test_dense_mask_index_cpu ", the inductor generated code is:
```c++
extern "C" void kernel(const float* __restrict__ in_ptr0,
                       const float* __restrict__ in_ptr1,
                       float* __restrict__ out_ptr0,
                       const long ks0)
{
    {
        {
            float tmp3 = 0;
            #pragma omp parallel num_threads(8)
            {
                #pragma omp for reduction(+:tmp3)
                for(long i0=0; i0<ks0; ++i0)
                {
                    {
                        auto tmp0 = in_ptr0[i0];
                        auto tmp1 = in_ptr1[2];
                        auto tmp2 = tmp0 * tmp1;
                        tmp3 += tmp2;
                    }
                }
            }
            out_ptr0[0] = tmp3;
        }
    }
}

```
The **OMP reduction** cannot guarantee a fixed reduce order for sum. Actually, I can observe different results here while repeating execute this generated CPP code.
```
-251.1077880859375 inductor
-251.10780334472656 inductor
-251.10780334472656 inductor
-251.10775756835938 inductor
```

I also tried this
```
import torch
size = 100000
a  = torch.randn(size)
b = torch.randn(3)
for i in range(10):
    c = torch.zeros((1, ))
    order = torch.randperm(size)
    for j in range(size):
        c += (a[order[j]] * b[2])
    print(c.item())
```
The above code will be executed by only 1 thread and the only differences is order for sum, and will get output:
```
1.209582805633545
1.2103385925292969
1.2097705602645874
1.2099699974060059
1.2103685140609741
1.2103989124298096
1.210781216621399
1.210976243019104
1.2101386785507202
1.2098314762115479
```

For "test_expanded_reduction_cpu", the generated code does not contain "OMP reduce", it will be a fixed order for sum. But this order is different with aten sum.
I tried 
```python
import torch

d1 = 2
d2 = 197
d3 = 256

torch.manual_seed(20)
a = torch.randn(d1, d2, d3)
b = a.sum((0, 1))
print(b[0].item())

ref = torch.zeros(d3, )
for k in range(d3):
  b = a[:, :, k]
  for i in range(d1):
      for j in range(d2):
        ref[k] += b[i][j]
print(ref[0].item())

ref = torch.zeros(d3, )
for k in range(d3):
  b = a[:, :, k]
  for i in range(d2):
      for j in range(d1):
        ref[k] += b[j][i]
print(ref[0].item())
```
and it will get different results for all 3 cases. 

To conclusion:
 - All UTs contains "sum" might be able to fail like these two. It might caused by a random order introduced by OMP reduce, or just have a difference reduce order compare with aten sum.
 - I do not know whether there is exist a correct order for "sum". If yes, the outputs of sum should be exactly the same. If not,  we may make the tolerance larger (currently atol = 1e-5, rtol=1.3e-6). From the ut logs, the max diff just slightly higher than current tolerance.

```
2022-10-16T05:48:44.9647872Z Mismatched elements: 1 / 256 (0.4%)
2022-10-16T05:48:44.9648144Z Greatest absolute difference: 2.002716064453125e-05 at index (36,) (up to 1e-05 allowed)
2022-10-16T05:48:44.9648416Z Greatest relative difference: 4.907013266928203e-06 at index (36,) (up to 1.3e-06 allowed)

```
